server/src/routes/asiv/payment.ts

```diff
@@ -1,27 +1,33 @@
 import { Router, Request, Response } from "express";
 
-import { HttpStatusCode } from "../../utils";
+import { HttpStatusCode, FLORA_API_KEY, CLIENT_API_KEY } from "../../utils";
 import axios from "axios";
+import { ApiError, ApiErrorMessage } from "../../errors";
 
-const CLIENT_KEY = "CLIENT";
-const FLORA_KEY = "FLORA";
 const router: Router = Router();
 
 // POST request - generate payment url
 router.post("/request", async (req: Request, res: Response) => {
   const { API_KEY, pid, uid } = req.body;
-  if (API_KEY === undefined || API_KEY != CLIENT_KEY) {
-    return res
-      .status(HttpStatusCode.FORBIDDEN)
-      .json({ error: "Invalid API key" });
-  } else if (pid === undefined) {
-    return res
-      .status(HttpStatusCode.BAD_REQUEST)
-      .json({ error: "No pid field provided" });
-  } else if (uid === undefined) {
-    return res
-      .status(HttpStatusCode.BAD_REQUEST)
-      .json({ error: "No uid field provided" });
+  if (API_KEY !== CLIENT_API_KEY) {
+    throw new ApiError(
+      HttpStatusCode.FORBIDDEN,
+      ApiErrorMessage.AsivPayment.INVALID_API_KEY
+    );
+  }
```

Comment:
@Yonggiee to throw an error to the API callee, simply throw an `ApiError`, the `customErrorHandler` will handle sending of the status and message to the callee.